### PR TITLE
[WIP] Update Defense Mode's Item Lists.

### PR DESCRIPTION
**Pull request by Rod995**
#### Summary
```SUMMARY: Content "Updates Defense Mode item list."```

#### Purpose of change
Defense Mode's item groups were outdated, so I'm updating the lists, as well as implementing increasingly advanced weapons to be sold from the caravan.

#### Describe the solution
Adds newer items to item groups.
Makes me wonder if item groups could be added instead, to save line space...

#### Describe alternatives you've considered
None at the moment.

#### Additional context
Yes.
Still a work in progress, don't merge.
Renewal of #28331, since my commits don't show up there anymore.

### Sorting Melee by DPM

> The item list at cdda-trunk.chezzo.com calculates what a melee item is using `damage + cutting damage + to-hit bonus >= 7`

This is a chart representing the Caravan's tech progression, increased by the number of shop encounters that occur.

*Extremely Not Final!*

Tech Level | # of Caravan Visits | DPM Min | DPM Max
--------- | ----- | ----- | -------
Level I | 1 | 0 | 8 |
Level II | 5 | 8 | 10 |
Level III | 8 | 10 | 12 |
Level IV | 11 | 12 | 14 |
Level V | 14 | 14 | 16 |
Level VI | 17 | 16 | 20 |
Level VII | 19 | 20 | 24 |
Level VIII | 20 | 24 | 28 |
Level IX | 22 | 28 | 32 |
Level X | 23 | 32 | 36 |
Max | 30 | 36+ | ∞ |

#### Notes

- While the list is currently unsorted, I plan to sort the weapons using their damage per turn, with `to_hit` being a straight modifier to it.
- Instruments are included in the melee item list, due to their propensity to become weapons. :p
- Fake items are also included, since some of them make decent early-game weapons, in my experience.

### CODE REVIEW

File: data/json/itemgroups/defense_mode.json [hunk 1]
```diff
@@ -162,5 +663,239 @@
       { "item": "UPS_off" },
       { "item": "mininuke" }
     ]
+  },
+  {
+    "type": "item_group",
+    "id": "defense_caravan_bionics",
```
Comment (OrenAudeles) [inline data/json/itemgroups/defense_mode.json]: Does defense mode have an Autodoc available to install the bionics, or are they auto-installed on purchase?

File: data/json/itemgroups/defense_mode.json [hunk 2]
```diff
@@ -162,5 +663,239 @@
...
+      { "item": "bio_remote" },
+      { "item": "bio_cable" },
+      { "item": "bio_syringe" },
+      { "item": "bio_weight" },
+      { "item": "bio_shock_absorber" }
+    ]
+  },
+  {
+    "type": "item_group",
+    "id": "defense_caravan_books",
```
Comment (OrenAudeles) [inline data/json/itemgroups/defense_mode.json]: Does the player have enough downtime to read books to increase skill levels?

File: src/defense.cpp [hunk 3]
```diff
@@ -1176,7 +1182,40 @@ std::vector<itype_id> caravan_items( caravan_category cat )
             return ret;
 
         case CARAVAN_MELEE:
-            item_list = item_group::items_from( "defense_caravan_melee" );
+            if( ( caravan_tech_level == 1 ) ){
```
Comment (OrenAudeles) [inline src/defense.cpp]: Don't need the parenthesis inside of the condition, can instead be `if( caravan_tech_level == 1 )` for each of these

File: data/json/itemgroups/defense_mode.json (hunk 1 above)
Comment (Rod995) [reply data/json/itemgroups/defense_mode.json]: I'll have to check, but most don't have an Autodoc available. And the ones that might are too big to actually search entirely, without the hardcoded limits stopping you.
Could always use the manual bionic installation mod, although that isn't the best workaround.

File: src/defense.cpp (hunk 3 above)
Comment (Rod995) [reply src/defense.cpp]: Fixed.

**Comment by Rod995** [comment]
Jenkins rebuild

**Comment by Rod995** [comment]
Rebasing broke the stupid PR, smh.

**Pull request closed without merging**